ivy/functional/frontends/tensorflow/general_functions.py

```diff
@@ -80,6 +81,23 @@ def fill(dims, value, name=None):
     return ivy.full(dims, value)
 
 
+@to_ivy_arrays_and_back
+def foldl(fn, elems, initializer=None):
+    try:
+        iter(elems)
+    except TypeError:
+        raise TypeError("elems must be an iterable object")
+
+    arr = ivy.array(elems)
+
+    if initializer is None and len(arr) > 0:
```

Comment:
Can we use something other than `len(arr)`? I tested with an `ivy.Array` containing a `tf.Variable` and this failed.